Mathlib/CategoryTheory/Sites/Whiskering.lean

```diff
@@ -35,14 +35,49 @@ variable {A : Type u₂} [Category.{v₂} A]
 
 variable {B : Type u₃} [Category.{v₃} B]
 
-variable {J : GrothendieckTopology C}
+variable (J : GrothendieckTopology C)
 
 variable {U : C} (R : Presieve U)
 
 variable (F G H : A ⥤ B) (η : F ⟶ G) (γ : G ⟶ H)
 
+/-- Describes the property of a functor to "preserve sheaves". -/
+class Functor.HasSheafCompose : Prop where
+  /-- For every sheaf `P`, `P ⋙ F` is a sheaf. -/
+  isSheaf (P : Cᵒᵖ ⥤ A) (hP : Presheaf.IsSheaf J P) : Presheaf.IsSheaf J (P ⋙ F)
+
+variable [F.HasSheafCompose J] [G.HasSheafCompose J] [H.HasSheafCompose J]
+
```

Comment:
```suggestion
class GrothendieckTopology.HasSheafCompose : Prop where
  /-- For every sheaf `P`, `P ⋙ F` is a sheaf. -/
  isSheaf (P : Cᵒᵖ ⥤ A) (hP : Presheaf.IsSheaf J P) : Presheaf.IsSheaf J (P ⋙ F)

variable [F.GrothendieckTopology J] [G.GrothendieckTopology J] [H.GrothendieckTopology J]

```

Reply:
You mean `variable [J.HasSheafCompose F]` etc, right?